.github/workflows/continuous_deployment.yml

```
@@ -182,74 +190,48 @@ jobs:
-      with:
-        python-version: 3.7
-    - name: Install dependencies
-      run: |
-          python -m pip install --upgrade "pip<20"
-          pip install -r requirements-dev.txt
-    - name: Train Model
-      working-directory: ${{ github.workspace }}
-      run: |
-          rasa train
-    - name: Set model name from Rasa version
+    - name: Download model from storage bucket
       run: |
-          python -c "import rasa; open('rasaversion.txt','w+').write(rasa.__version__)"
-          rasa_version=`cat rasaversion.txt`
-          model_path=`ls models/*.tar.gz | head -n 1`
-          model_timestamp=$(basename "$model_path" .tar.gz)
-          model_name="$model_timestamp"_rasa"$rasa_version"
-          renamed_model_path=models/"$model_name".tar.gz
-          mv $model_path $renamed_model_path
-          echo "MODEL_NAME=${model_name}" >> $GITHUB_ENV
-          echo "MODEL_PATH=${renamed_model_path}" >> $GITHUB_ENV
-
-    - uses: google-github-actions/setup-gcloud@master
-      name: Authenticate with gcloud 🎫
-      with:
-        version: "${{ env.GCLOUD_VERSION }}"
-        service_account_email: ${{ env.SARA_GKE_SERVICE_ACCOUNT_NAME }}
-        service_account_key: ${{ secrets.SARA_GKE_SERVICE_ACCOUNT_KEY }}
+        # Fetch the latest model based on when it created
+        MODEL_URL=$(gsutil ls -l ${{ secrets.STORAGE_BUCKET_URL }}/rasa_demo_models/ | grep "tar.gz" | grep $RASA_VERSION | sort -k2 | tail -n1 | awk '{print $3}')
+        # Get the model local path, get rid of bucket rul
+        MODEL_PATH=$(basename $MODEL_URL)
+        # Download the model from bucket
+        gsutil -m cp ${MODEL_URL} ${MODEL_PATH}
 
-    - name: Upload model to storage bucket
-      run: |
-          gsutil cp "${MODEL_PATH}" ${STORAGE_BUCKET_URL}/rasa_demo_models
+        echo "MODEL_PATH=${MODEL_PATH}" >> $GITHUB_ENV
+        echo "MODEL_NAME=$(basename $MODEL_PATH .tar.gz)" >> $GITHUB_ENV
 
     - name: Upload model to Rasa X
       working-directory: ${{ github.workspace }}
+      timeout-minutes: 10
       run: |
-           auth_token=$(curl --request POST \
-              --url https://${RASA_X_DOMAIN}/api/auth \
-              --header 'Content-Type: application/json' \
-              --data "{
-              \"username\": \"${RASA_X_USERNAME}\",
-              \"password\": \"${RASA_X_PASSWORD}\"
-            }" | jq -r '.access_token')
-
-            echo "AUTH_TOKEN=${auth_token}" >> $GITHUB_ENV
-
-            curl -k -F "model=@${MODEL_PATH}" \
-              --url https://$RASA_X_DOMAIN/api/projects/default/models \
-              --header "Authorization: Bearer ${auth_token}"
+          auth_token=$(curl -s -X POST \
+            -H 'Accept: application/json' \
+            -H 'Content-Type: application/json' \
+            --data '{\"username\": \"${RASA_X_USERNAME}\",\"password\": \"${RASA_X_PASSWORD}\"}' \
+            https://${{ env.RASA_X_DOMAIN }}/api/auth | tee /tmp/output_auth.txt | jq -r '.access_token')
```

I think the issue is the quoting - the previous side works, this one doesn't locally.
```suggestion
          auth_token=$(curl -s -X POST \
            -H 'Accept: application/json' \
            -H 'Content-Type: application/json' \
            --data "{\"username\": \"${RASA_X_USERNAME}\",\"password\": \"${RASA_X_PASSWORD}\"}" \
            https://${{ env.RASA_X_DOMAIN }}/api/auth | tee /tmp/output_auth.txt | jq -r '.access_token')
```

---

Always quotation marks 😞 
Let me try 🏃🏻